### PR TITLE
fix insert above shortcut

### PR DESCRIPTION
**Pull request by alex-Arc**
The big change is the unification of the 2 separate cursor stores
It would be good try to break it by clicking all over, so far i've not found any weird behaviour

### CODE REVIEW

File: apps/client/src/features/rundown/Rundown.tsx [hunk 1]
```diff
@@ -92,23 +105,26 @@ export default function Rundown({ data }: RundownProps) {
...
+        newCursor =
+          (direction === 'up' ? getLastNormal(rundown, order)?.id : getFirstNormal(rundown, order)?.id) ?? null;
+        newIndex = direction === 'up' ? order.length : 0;
       } else {
         // otherwise we select the next or previous
-        newCursor =
-          direction === 'up'
-            ? getPreviousNormal(rundown, order, cursor).entry?.id
-            : getNextNormal(rundown, order, cursor).entry?.id;
+        const selecteted =
```
Comment (cpvalente) [inline apps/client/src/features/rundown/Rundown.tsx:117]: Mind the typo `selected`

File: apps/client/src/features/rundown/Rundown.tsx [hunk 2]
```diff
@@ -44,39 +47,49 @@ export default function Rundown({ data }: RundownProps) {
...
-    (type: SupportedEvent | 'clone', cursor: string | null, above = false) => {
-      const adjustedCursor = above ? getPreviousNormal(rundown, order, cursor ?? '').entry?.id ?? null : cursor;
+  const insertCopyAtId = useCallback(
+    (atId: string | null, copyId: string | null, above = false) => {
+      const adjustedCursor = above ? getPreviousNormal(rundown, order, atId ?? '').entry?.id ?? null : atId;
+      if (copyId === null) {
+        // we cant clone without selection
+        return;
+      }
+      const cloenEntry = rundown[copyId];
```
Comment (cpvalente) [inline apps/client/src/features/rundown/Rundown.tsx:66]: Mind typo `cloneEntry`

**Review by cpvalente — APPROVED** [review]
Code looks good, just a few cleanup steps

I didn't have a chance to test it but trust that everything is fine

File: apps/client/src/features/rundown/Rundown.tsx [hunk 3]
```diff
@@ -44,39 +47,49 @@ export default function Rundown({ data }: RundownProps) {
...
-          addEvent(newEvent);
-        }
-      } else if (type === SupportedEvent.Event) {
+      if (type === SupportedEvent.Event) {
         const newEvent = {
           type: SupportedEvent.Event,
         };
         const options = {
           after: adjustedCursor,
+          //TODO: this might be incorrect
```
[the remark this answers is not in the document]
Comment (alex-Arc) [reply apps/client/src/features/rundown/Rundown.tsx:92]: was unsure what lastEventId ment, but looks to be for linking so it is the same as the after?

**Review by cpvalente — APPROVED** [review]
Good work

**Pull request merged**